geode-apis-compatible-with-redis/src/integrationTest/java/org/apache/geode/redis/internal/executor/sortedset/AbstractZIncrByIntegrationTest.java

```diff
@@ -0,0 +1,290 @@
+    jedis.sendCommand(STRING_KEY, Protocol.Command.ZINCRBY, STRING_KEY, "-Infinity", STRING_MEMBER);
+    assertThat(jedis.zscore(STRING_KEY, STRING_MEMBER)).isEqualTo(NEGATIVE_INFINITY);
+  }
+
+  @Test
+  public void nullRedisSetShouldCreateNonExistentMember_withIncrementOfPositiveInfinity() {
+    final double increment = POSITIVE_INFINITY;
+    jedis.zadd(KEY, 0.0, "something".getBytes()); // init the key but not the member
+
+    jedis.zincrby(KEY, increment, MEMBER);
+    assertThat(jedis.zscore(KEY, MEMBER)).isEqualTo(increment);
+  }
+
+  @Test
+  public void nullRedisSetShouldCreateNonExistentMember_withIncrementOfNegativeInfinity() {
+    final double increment = NEGATIVE_INFINITY;
+    jedis.zadd(KEY, 0.0, "something".getBytes()); // init the key but not the member
+    jedis.zincrby(KEY, increment, MEMBER);
+    assertThat(jedis.zscore(KEY, MEMBER)).isEqualTo(increment);
+  }
+
+  @Test
+  public void shouldCreateNonExistentKeyWithInfiniteScore_whenIncrementIsPositiveInfinity() {
+    final double increment = POSITIVE_INFINITY;
+    jedis.zincrby(KEY, increment, MEMBER);
+    assertThat(jedis.zscore(KEY, MEMBER)).isEqualTo(increment);
+  }
+
+  @Test
+  public void shouldCreateNonExistentKeyWithInfiniteScore_whenIncrementIsNegativeInfinity() {
+    final double increment = NEGATIVE_INFINITY;
+    jedis.zincrby(KEY, increment, MEMBER);
+    assertThat(jedis.zscore(KEY, MEMBER)).isEqualTo(increment);
+  }
+
+  @Ignore("blocked by GEODE-9317")
+  @Test
+  public void scoreOfPositiveInfinityShouldRemainInfinite_whenIncrementingOrDecrementing() {
+    jedis.zadd(KEY, POSITIVE_INFINITY, MEMBER);
+
+    jedis.zincrby(KEY, 100.0, MEMBER);
+    assertThat(jedis.zscore(KEY, MEMBER)).isEqualTo(POSITIVE_INFINITY);
+
+    jedis.zincrby(KEY, -1000.0, MEMBER);
+    assertThat(jedis.zscore(KEY, MEMBER)).isEqualTo(POSITIVE_INFINITY);
+  }
+
+  @Ignore("blocked by GEODE-9317")
+  @Test
+  public void scoreOfNegativeInfinityShouldRemainInfinite_whenIncrementingOrDecrementing() {
+    jedis.zadd(KEY, NEGATIVE_INFINITY, MEMBER);
+
+    jedis.zincrby(KEY, 100.0, MEMBER);
+    assertThat(jedis.zscore(KEY, MEMBER)).isEqualTo(NEGATIVE_INFINITY);
+
+    jedis.zincrby(KEY, -1000.0, MEMBER);
+    assertThat(jedis.zscore(KEY, MEMBER)).isEqualTo(NEGATIVE_INFINITY);
+  }
+
+  @Ignore("blocked by GEODE-9317")
+  @Test
+  public void incrementingScoreOfInfinityByNegativeInfinity_throwsNaNError() {
+    jedis.zadd(KEY, POSITIVE_INFINITY, MEMBER);
+
+    assertThatThrownBy(() -> jedis.zincrby(KEY, NEGATIVE_INFINITY, MEMBER))
+        .hasMessageContaining(RedisConstants.ERROR_OPERATION_PRODUCED_NAN);
+  }
+
+  @Ignore("blocked by GEODE-9317")
+  @Test
+  public void incrementingScoreOfNegativeInfinityByPositiveInfinity_throwsNaNError() {
+    jedis.zadd(KEY, NEGATIVE_INFINITY, MEMBER);
+
+    assertThatThrownBy(() -> jedis.zincrby(KEY, POSITIVE_INFINITY, MEMBER))
+        .hasMessageContaining(RedisConstants.ERROR_OPERATION_PRODUCED_NAN);
+  }
```

Comment:
We don't actually need to use `ZADD` for these, you could do:
```
    jedis.zincrby(KEY, NEGATIVE_INFINITY, MEMBER);

    assertThatThrownBy(() -> jedis.zincrby(KEY, POSITIVE_INFINITY, MEMBER))
        .hasMessageContaining(RedisConstants.ERROR_OPERATION_PRODUCED_NAN);
 ```

Reply:
good point. but I'm wrapping up the zadd work that will enable these tests. I'd rather just wait for that since it'll read a little clearer

Reply:
good point